canvas_sdk/v1/data/base.py

```diff
@@ -12,3 +21,38 @@ def committed(self) -> "models.QuerySet":
 
     def for_patient_id(self, patient_id: str) -> "models.QuerySet":
         return self.filter(patient__id=patient_id)
+
+
+class ValueSetLookupQuerySet(models.QuerySet):
+    def in_value_set(
+        self, value_set: Type["ValueSet"], code_system_mapping: dict[str, str] | None = None
+    ) -> models.QuerySet:
+        """
+        Filters conditions, medications, etc. to those found in the inherited ValueSet class that is passed.
+
+        For example:
+
+        from canvas_sdk.v1.data.condition import Condition
+        from canvas_sdk.value_set.v2022.condition import MorbidObesity
+        morbid_obesity_conditions = Condition.objects.in_value_set(MorbidObesity)
+
+        This method can also be chained like so:
+
+        Condition.objects.in_value_set(MorbidObesity).in_value_set(AnaphylacticReactionToCommonBakersYeast)
+
+        A code/system mapping dictionary can be optionally passed to define value_set codes and URI values
+        so that plugin authors can use custom value sets.
```

Comment:
I'm not sure I follow this use case. Why wouldn't a plugin developer just subclass `ValueSet` and have it be treated the same as any other?

Reply:
They would subclass `ValueSet`, but each `ValueSet` class defines the codes by system as attribute variable names (i.e. `ICD10CM = {"G04.32"}`). There needs to be a way to 1) define which attribute variable names in the class are valid codes and 2) map that to a sytem value (i.e. `http://www.nlm.nih.gov/research/umls/rxnorm`) to be looked up in the database.

Reply:
Looking at this again, I think we can have developers put the system/code mapping in the ValueSet itself. It would just have to be named something specific. The Canvas ValueSets wouldn't contain it, and in that case, we can just use the default.

Reply:
The mapping between the constant name and its representative system URI should be part of the `ValueSet` base class.

See the workflow kit:
- ValueSet defines the list of valid code systems
  - https://github.com/canvas-medical/canvas-workflow-kit/blob/65349e7cb3aadfa84f39a84c4f609a2399eff1ed/canvas_workflow_kit/value_set/value_set.py#L59
- `PatientRecordSet` class defines the constants used when filtering
  - https://github.com/canvas-medical/canvas-workflow-kit/blob/65349e7cb3aadfa84f39a84c4f609a2399eff1ed/canvas_workflow_kit/patient_recordset.py#L20-L61


Plugin developers providing their own `ValueSets` should not be required to do any mapping of constants to system URIs. They should be able to define a ValueSet class just as concisely as we do and be able to use it identically to the built-in ones.

Reply:
Yep, that's what I was thinking of doing.

Reply:
I added the constants to the base classes.